pkg/webhook/controlplaneexposure/ensurer.go

```diff
@@ -87,27 +88,35 @@ func (e *ensurer) EnsureKubeAPIServerDeployment(ctx context.Context, ectx generi
 	return nil
 }
 
-// EnsureETCDStatefulSet ensures that the etcd stateful sets conform to the provider requirements.
-func (e *ensurer) EnsureETCDStatefulSet(ctx context.Context, ectx genericmutator.EnsurerContext, ss *appsv1.StatefulSet) error {
-	e.ensureVolumeClaimTemplates(&ss.Spec, ss.Name)
-	return nil
-}
-
-func (e *ensurer) ensureVolumeClaimTemplates(spec *appsv1.StatefulSetSpec, name string) {
-	t := e.getVolumeClaimTemplate(name)
-	spec.VolumeClaimTemplates = extensionswebhook.EnsurePVCWithName(spec.VolumeClaimTemplates, *t)
-}
-
-func (e *ensurer) getVolumeClaimTemplate(name string) *corev1.PersistentVolumeClaim {
+// EnsureETCD ensures that the etcd conform to the provider requirements.
+func (e *ensurer) EnsureETCD(ctx context.Context, ectx genericmutator.EnsurerContext, etcd *druidv1alpha1.Etcd) error {
 	var (
-		etcdStorage             config.ETCDStorage
-		volumeClaimTemplateName = name
+		etcdStorage config.ETCDStorage
 	)
 
-	if name == v1beta1constants.ETCDMain {
+	capacity := resource.MustParse("10Gi")
+	class := ""
+	if etcd.Name == v1beta1constants.ETCDMain {
 		etcdStorage = *e.etcdStorage
-		volumeClaimTemplateName = controlplane.EtcdMainVolumeClaimTemplateName
+		if etcdStorage.Capacity != nil {
+			capacity = *etcdStorage.Capacity
+		}
+		if etcdStorage.ClassName != nil {
+			class = *etcdStorage.ClassName
+		}
 	}
 
-	return controlplane.GetETCDVolumeClaimTemplate(volumeClaimTemplateName, etcdStorage.ClassName, etcdStorage.Capacity)
+	if etcd.Name == v1beta1constants.ETCDMain && e.etcdStorage != nil {
```

Comment:
You could remove line 99-107 in favour of this block

Reply:
yup done.